### DISCUSSION TITLE
Different form schema for edit and view

### DISCUSSION THREAD

**Discussion by stefanolissa** [filamentphp/filament#6053] [General]
Hi,

I'm just trying in a resource class to get the form() method to build a different schema for the view and edit page. Is that possible? How can I distinguish if the view or the edit page is going to be shown?

Thank you.

**Comment by ManojKiranA** [comment]
Is that resource or Relation manager ?

**Comment by stefanolissa** [comment]
It's a resource, but I now found that every page for a resource has the getFormSchema() that can be used to define the schema you need for your form.

If I'm understanding the code correctly, if you do not define that method the resource class form() method if used.

I need to study this topic more deeply... especially for EditRecord and ViewRecord classes.

Thank you!

**Reply by danharrin** [reply]
You are correct, you can override that method on a page to provide a different form
